### PR TITLE
Analytic requests

### PR DESCRIPTION
**Pull request by hermantretteteig**
- Fetch analytic for each climate element and population
- Moving from GeoJSON to only orgUnits hierarchy information (agreed with Knut)
- Possible to select several orgUnits or an entire level
- Some files changed to TypeScript

### CODE REVIEW

File: src/components/prediction/PredictionPage.jsx [hunk 1]
```diff
@@ -13,12 +13,13 @@ const defaultPeriod = {
 };
 
 const PredictionPage = () => {
-  const [predictionTarget, setPredictionTarget] = useState();
-  const [populationData, setPopulationData] = useState();
+  const [predictionTarget, setPredictionTarget] = useState({displayName: 'IDSR Malaria', id: 'vq2qO3eTrNi'});
```
Comment (hermantretteteig) [inline src/components/prediction/PredictionPage.jsx:16]: I defined some default values here, just temporary as we develop.

File: src/components/prediction/MonthlyPeriodSelect.jsx [hunk 2]
```diff
@@ -26,8 +26,8 @@ const MonthlyPeriodSelect = ({ period, onChange }) => {
 };
 
 MonthlyPeriodSelect.propTypes = {
-  currentPeriod: PropTypes.object.isRequired,
-  onUpdate: PropTypes.func.isRequired,
+  period: PropTypes.object.isRequired,
```
Comment (hermantretteteig) [inline src/components/prediction/MonthlyPeriodSelect.jsx:29]: I think these ones, were named wrongly. Changed it.

File: src/Globals.d.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+declare module "*.module.css";
+declare module "*.module.scss";
```
Comment (turban) [inline src/Globals.d.ts:2]: Bruker vi Sass (.scss)?

File: src/components/AboutPage.jsx [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import i18n from "@dhis2/d2-i18n";
 import styles from "./styles/AboutPage.module.css";
+import useAnalyticRequest from "../hooks/useAnalyticRequest";
```
Comment (turban) [inline src/components/AboutPage.jsx:3]: Not in use on this page

File: src/components/prediction/DownloadData.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  setStartDownload: (startDownload : boolean) => void
+}
+
+const DownloadData = ({ period, setStartDownload, orgUnitLevels, orgUnits, temperatureData, precipitationData, populationData}: DownloadDataProps) => {
+  
+  //Concat selected orgUnits (either national, a district, chiefdom or facility) with the id of selected levels (districts, chiefdoms, facilities)
+  const mergedOrgUnits = orgUnitLevels.map(level => "LEVEL-"+level).join(";")+";"+orgUnits.map((ou : any) => ou.id).join(";")
+
+  //Convert the period selected to a DHIS2-standard list of months
+  const fillPeriodesMonths = (period : any) => {
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:25]: This is fine for now, just remember that we need to support multiple period types in the future.

File: src/components/prediction/DownloadData.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  const { orgUnits: allOrgUnits } = useOrgUnits();
+
+  const downloadZip = ()  => {
+    const zip = new JSZip();
+    //Add orgUnits to zip
+    zip.file("orgUnits.json", JSON.stringify(allOrgUnits))
+    zip.file("precipitation.json", JSON.stringify(precipitation))
+    zip.file("population.json", JSON.stringify(population))
+    zip.file("temperature.json", JSON.stringify(temperature))
+
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:56]: Add "orgUnits.geojson" (also fine to add them back in a later PR)

File: src/components/prediction/DownloadData.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  const {data : precipitation} = useAnalyticRequest(precipitationData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const {data : population} = useAnalyticRequest(populationData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const {data : temperature} = useAnalyticRequest(temperatureData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const { orgUnits: allOrgUnits } = useOrgUnits();
+
+  const downloadZip = ()  => {
+    const zip = new JSZip();
+    //Add orgUnits to zip
+    zip.file("orgUnits.json", JSON.stringify(allOrgUnits))
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:52]: Fine for now. For later we should not add org units that are not specified by the user (for building trust in the data that is being shared).

File: src/components/prediction/OrgUnitTree.tsx [hunk 8]
```diff
@@ -3,32 +3,39 @@ import i18n from "@dhis2/d2-i18n";
 import { useEffect } from "react";
 import { OrganisationUnitTree } from "@dhis2/ui";
 import useOrgUnitRoots from "../../hooks/useOrgUnitRoots";
+import React from "react";
```
Comment (turban) [inline src/components/prediction/OrgUnitTree.tsx:6]: I don't think this import is needed.

**Review by turban — CHANGES_REQUESTED** [review]

File: src/components/prediction/OrgUnitTree.tsx (hunk 8 above)
Comment (hermantretteteig) [reply src/components/prediction/OrgUnitTree.tsx:6]: I think TypeScript requires it, get an error if I remove it

File: src/Globals.d.ts (hunk 3 above)
Comment (hermantretteteig) [reply src/Globals.d.ts:2]: Niks, bare copy-paste fra StackOverflow

File: src/components/prediction/OrgUnitTree.tsx (hunk 8 above)
Comment (turban) [reply src/components/prediction/OrgUnitTree.tsx:6]: Ok, could be added as:

`import React, { useEffect } from "react";`

For DHIS2 projects we include external imports before internal ones.

**Pull request merged**